src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -1151,7 +1151,8 @@ _mongoc_server_description_equal (mongoc_server_description_t *sd1, mongoc_serve
       return false;
    }
 
-   if (0 != memcmp (&sd1->error, &sd2->error, sizeof (bson_error_t))) {
+   /* error may have uninitialized bytes after terminating '\0' */
+   if (0 != strncmp (&sd1->error, &sd2->error, sizeof (bson_error_t))) {
```

Comment:
```suggestion
   if (sd1->error.domain != sd2->error.domain || sd1->error.code != sd2->error.code ||
       0 != strncmp (sd1->error.message, sd2->error.message, BSON_ERROR_BUFFER_SIZE)) {
```
Recommend doing proper logical field comparison of `bson_error_t` rather than attempting to shortcut with byte-comparison (as done prior with `memcmp` and with the proposed `strncmp`). The proposed use of `strncmp` will incorrectly suspend comparison if there is any null byte in the `domain` or `code` fields before it has a chance to reach the `message` field.